Marlin/src/gcode/geometry/G92.cpp

```diff
@@ -93,10 +93,37 @@ void GcodeSuite::G92() {
       COPY(coordinate_system[active_coordinate_system], position_shift);
   #endif
 
-  bool didA = parser.seenval('A');
-  if (didA) {
-    const float curr_z = current_position[Z_AXIS];
-    const float adjust_z = parser.value_axis_units(Z_AXIS);
+  bool doZAdjust = false;
+  float adjust_z = 0.0;
+
+
+
+  if(parser.seenval('A')){
+    adjust_z = parser.value_axis_units(Z_AXIS);
+    doZAdjust = true;
+  }else if(parser.seenval('B')){
+    float final_z = parser.value_axis_units(Z_AXIS);
+
+
+    SERIAL_ECHO_START();
+    SERIAL_ECHOLNPAIR("final_z: ", final_z);
+    SERIAL_ECHOLNPAIR("primaryZTO: ", primaryZTO);
+    SERIAL_ECHOLNPAIR("secondaryZTO: ", secondaryZTO);
+
+
+
+    if(active_extruder == 0)
+      adjust_z = final_z - primaryZTO;
+    else
+      adjust_z = final_z - secondaryZTO;
+    doZAdjust = true;
```

Comment:
Code-style/formatting - this should be one of the first statements in this block so that it is immediately visible when reading the code and not mistaken for being conditional.